packages/contracts/src/contracts/current/protocol/AssetTransferProxy/AssetTransferProxy.sol

```diff
@@ -0,0 +1,132 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.4.21;
+
+import "./AssetProxyEncoderDecoder.sol";
+import "./IAssetProxy.sol";
+import "../../utils/Authorizable/Authorizable.sol";
+
+contract AssetTransferProxy is
+    AssetProxyEncoderDecoder,
+    Authorizable
+{
+    // Mapping from Asset Proxy Id's to their respective Asset Proxy
+    mapping (uint8 => IAssetProxy) public assetProxies_;
+
+    // Logs registration of new asset proxy
+    event LogAssetProxyRegistration(
+        AssetProxyId id,
+        address assetClassAddress,
+        bool overwrite,
+        bool did_overwrite
+    );
+
+    // Logs unregistration of an existing asset proxy
+    event LogAssetProxyUnregistration(
+        AssetProxyId id,
+        address assetClassAddress
+    );
+
+    /// @dev Delegates transfer to the corresponding asset proxy.
+    /// @param assetMetadata Byte array encoded for the respective asset proxy.
+    /// @param from Address to transfer token from.
+    /// @param to Address to transfer token to.
+    /// @param amount Amount of token to transfer.
+    /// @return Success of transfer.
+    function transferFrom(
+        bytes assetMetadata,
+        address from,
+        address to,
+        uint256 amount)
+        public
+        onlyAuthorized
+        returns (bool)
+    {
+        // Lookup asset proxy
+        require(assetMetadata.length >= 1);
+        uint8 id = uint8(assetMetadata[0]);
+        IAssetProxy assetProxy = assetProxies_[id];
+        require(assetProxy != address(0x0));
+
+        // Delegate transfer to asset proxy
+        return assetProxy.transferFrom(assetMetadata, from, to, amount);
```

Comment:
Using a byte array allows us to encode arbitrary metadata. For ERC20 all we need is a the id and address, but this could be different for other standards.